fabric-screen-handler-api-v1/src/main/java/net/fabricmc/fabric/api/screenhandler/v1/OverwritingScreenHandlerFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.screenhandler.v1;
+
+import net.minecraft.screen.NamedScreenHandlerFactory;
+import net.minecraft.screen.ScreenHandlerFactory;
+import net.minecraft.screen.SimpleNamedScreenHandlerFactory;
+import net.minecraft.text.Text;
+
+/**
+ * A {@link net.minecraft.screen.SimpleNamedScreenHandlerFactory} that
+ * "overwrites" the current screen without closing it.
+ * In vanilla game, opening a new screen will always send the close screen packet.
+ * This, among other things, causes the mouse cursor to move to the center of the screen,
+ * which might not be expected in some cases. Opening a screen handler with this factory
+ * skips sending the packet and "overwrites" the current screen, preventing the cursor
+ * movement.
+ */
+public final class OverwritingScreenHandlerFactory extends SimpleNamedScreenHandlerFactory {
```

Comment:
I'd remove this class. Modders already need to implement `ScreenHandlerFactory`, they can override the additional function if they want to. Since this is interface-injected, they should also be able to use the following pattern:
```java
new SimpleNamedScreenHandlerFactory(whatever) {
    @Override
    public boolean shouldCloseCurrentScreen() {
        return false;
    }
};
```
Slightly more verbose, but it saves us from this extra API class.

Reply:
Sounds good.